File: 01-数据结构/01-数组/go/LC 验证回文串.go
package main

import (
	"strings"
)

func isPalindrome(s string) bool {
    s = strings.ToUpper(s)
    chs := []rune{}
    for _, v := range s {
        if v >= 'A' && v <= 'Z' || v >= '0' && v <= '9' {
            chs = append(chs, v)
        }
    }
    l, r := 0, len(chs)-1

    for l < r {
        if chs[l] != chs[r] {
            return false
        }
        l++
        r--
    }

    return true
}

File: 01-数据结构/01-数组/c++/LC 验证回文串.cpp
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {
        vector<char> chs;
        for (auto x : s) {
            if (x >= 'a' && x <= 'z' || x >= 'A' && x <= 'Z' || x >= '0' && x <= '9') {
                if (x >= 'A' && x <= 'Z') {
                    x = x - 'A' + 'a';
                }
                chs.push_back(x);
            }
        }

        
        int l = 0;
        int r = chs.size()-1;
        while (l < r) {
            if (chs[l] != chs[r]) {
                return false;
            }
            l++;
            r--;
        }
        return true;
    }
};